assets/js/sharedClientFunctions.js

```diff
@@ -80,11 +81,10 @@ module.exports = {
       });
     };
     $scope.deleteRef = function (id) {
-      var current_user = $scope.refUser || $scope.user;
       var url = "/reference/delete";
       io.socket.post(url, {refId: id}, function (data, res) {
         if (res.statusCode === 200) {
-          current_user.references = current_user.references.filter(function (ref) {
+          $scope[current_user].references = $scope[current_user].references.filter(function (ref) {
```

Comment:
Basically these functions used to exist in two separate locations (one in refCtrl and one in userCtrl). I combined them when refactoring to avoid duplicate code, but that requires the parameters to be slightly more complicated